hummingbot/hummingbot
# Fix/solve bitmart issues

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
Update Bitmart connector to the latest connector standards. The intention is to solve all problems that were affecting it.
Unit tests are updated in the PR too


**Tests performed by the developer**:
Run a pure market making strategy using the connector.


**Test summary**:
    * Clone and install the feature branch without issue
    * Connect the API keys and check the balance
    * Create and run pure mm strategy and started the strategy without issue
    * Order creation and cancelation: OK
    * order_book live: Ok
    * Status live: OK
    * Ticker live: Ok
    * Filled trade event
        * Recording of filled orders: OK
        * Order delay: Ok
        * History: Ok
        * Fees paid: Ok
    * Test fast and long refresh duration
    * Multiple levels: Ok


**Tips for QA testing**:
Full validation of the connector.
Review any open issue for Bitmart and check the problem is no longer present in this PR

This PR comes from: https://github.com/CoinAlpha/hummingbot/pull/257

PRP: https://snapshot.org/#/hbot-prp.eth/proposal/0x4ce5fc01d3bd0d4559d2256bfaa55cf8aca4957556322b62bd9ee4fa35482021
